lib/saml_idp/request.rb

```diff
@@ -5,12 +5,8 @@ class Request
     def self.from_deflated_request(raw)
       if raw
         decoded = Base64.decode64(raw)
-        zstream = Zlib::Inflate.new(-Zlib::MAX_WBITS)
         begin
-          inflated = zstream.inflate(decoded).tap do
```

Comment:
It looks like it can remain the old way in some regard? http://ruby-doc.org/stdlib-2.1.1/libdoc/zlib/rdoc/Zlib.html#method-c-inflate if you find that it cannot remain the old way for ` > 2.1` we should check `respond_to` so we can handle both. I'm betting we still have some people using 1.9-2.1